mmv1/products/vertexai/Schedule.yaml

```diff
@@ -0,0 +1,323 @@
+                description: |
+                  The display name of the Pipeline. The name can be up to 128 characters long
+                  and can consist of any UTF-8 characters.
+              - !ruby/object:Api::Type::NestedObject
+                name: pipelineSpec
+                description: The spec of the pipeline.
+                required: true
+                # TODO: add pipelineSpec properties
+              - !ruby/object:Api::Type::KeyValuePairs
+                name: labels
+                description: |
+                  The labels with user-defined metadata to organize PipelineJob.
+                  Label keys and values can be no longer than 64 characters
+                  (Unicode codepoints), can only contain lowercase letters,
+                  numeric characters, underscores and dashes. International
+                  characters are allowed. See https://goo.gl/xmQnxf for more
+                  information and examples of labels.
+              - !ruby/object:Api::Type::NestedObject
+                name: runtimeConfig
+                description: Runtime config of the pipeline.
+                required: true
+                properties:
+                  # TODO parameters
+                  - !ruby/object:Api::Type::String
+                    name: gcsOutputDirectory
+                    description: |
+                      Required. A path in a Cloud Storage bucket, which will be treated as the root
+                      output directory of the pipeline. It is used by the system to generate the paths
+                      of output artifacts. The artifact paths are generated with a sub-path pattern
+                      {job_id}/{taskId}/{output_key} under the specified output directory.
+                      The service account specified in this pipeline must have the storage.objects.get
+                      and storage.objects.create permissions for this bucket.
+                    required: true
+                  # TODO parameterValues
+                  - !ruby/object:Api::Type::Enum
+                    name: failurePolicy
+                    description: |
+                      Represents the failure policy of a pipeline. Currently, the default of a pipeline
+                      is that the pipeline will continue to run until no more tasks can be executed,
+                      also known as PIPELINE_FAILURE_POLICY_FAIL_SLOW. However, if a pipeline is set
+                      to PIPELINE_FAILURE_POLICY_FAIL_FAST, it will stop scheduling any new tasks when
+                      a task has failed. Any scheduled tasks will continue to completion.
+                    values:
+                      - :PIPELINE_FAILURE_POLICY_UNSPECIFIED
+                      - :PIPELINE_FAILURE_POLICY_FAIL_SLOW
+                      - :PIPELINE_FAILURE_POLICY_FAIL_FAST
+                    default_value: :PIPELINE_FAILURE_POLICY_UNSPECIFIED
+                  # TODO inputArtifacts
```

Comment:
Not sure how to implement this type in magic modules
`map (key: string, value: object (InputArtifact))`

https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.pipelineJobs#PipelineJob.RuntimeConfig

Reply:
Could you try to have a key-value pair with a nested object?

Reply:
It looks like key-value pairs are only for string values? https://googlecloudplatform.github.io/magic-modules/docs/how-to/add-mmv1-resource/#keyvaluepairs-labels--annotations

Reply:
How about `!ruby/object:Api::Type::Map`? You can find some examples by searching it in mmv1